### PR TITLE
- switch from nginx to nginx unit

### PR DESCRIPTION
**Pull request by mtracz**
This PR implements Nginx Unit web server.
https://unit.nginx.org/

Due to lack of official Unit Docker image with php 8.3 module, we need to compile it on our own from the source.
- https://hub.docker.com/_/unit
- https://github.com/nginx/unit/blob/master/pkg/docker/Dockerfile.php8.2

Setup is working on local environment, but need to be checked on beta environment for some longer period of time.

---

Since Unit is based on API config, we need to reload config via curl. `make run` reload config automatically, but there is `make unit-reload-config` to do it manually.

Unit config might need to be adjusted. At this moment it is base config.

---

Additionally:
- bump pecl redis extension from 5.3.7 to 6.0.2
- bump composer image to 2.6.6
- bump node image to 21.4.0
- bump Xdebug to 3.3.0

### CODE REVIEW

**Review by krzysztofrewak — APPROVED** [review]

**Review by KarolZygadlo — APPROVED** [review]

**Review by EwelinaSkrzypacz — DISMISSED** [review]

File: environment/prod/app/nginx-unit.config.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        "requests": 1000
+      },
+      "processes": {
+        "max": 10,
+        "spare": 2,
+        "idle_timeout": 20
+      }
+    }
+  }
+}
```
Comment (kamilpiech97) [inline environment/prod/app/nginx-unit.config.json:38]:
```suggestion
}

```

File: Makefile [hunk 2]
```diff
@@ -32,7 +32,8 @@ build:
 	@docker compose --file ${DOCKER_COMPOSE_FILE} build --pull
 
 run:
-	@docker compose --file ${DOCKER_COMPOSE_FILE} up --detach
+	@docker compose --file ${DOCKER_COMPOSE_FILE} up --detach \
+	&& sleep 2 && make unit-reload-config
```
Comment (kamilpiech97) [inline Makefile:36]: For what is that `sleep 2`?

File: environment/dev/app/nginx-unit.config.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        "requests": 1000
+      },
+      "processes": {
+        "max": 10,
+        "spare": 1,
+        "idle_timeout": 20
+      }
+    }
+  }
+}
```
Comment (kamilpiech97) [inline environment/dev/app/nginx-unit.config.json:38]:
```suggestion
}

```

File: Makefile (hunk 2 above)
Comment (mtracz) [reply Makefile:36]: It's workaround to reload nginx unit server. Unit configuration base on API, so as you can see in `unit-reload-config` we are using curl.
```shell
unit-reload-config:
	@echo "Reloading Nginx Unit config..." \
	&& docker compose --file ${DOCKER_COMPOSE_FILE} exec ${DOCKER_COMPOSE_APP_CONTAINER} curl -X PUT --data-binary @/application/environment/dev/app/nginx-unit.config.json --unix-socket /var/run/control.unit.sock http://localhost/config
```

Unit needs a while to spin up. So we postpone this for 2 seconds. After compose up, unit is reloading config, so we can be sure that it's working with the current json config.

It's not the best solution but it's simple and it's working.

**Review by kamilpiech97 — APPROVED** [review]

**Review by EwelinaSkrzypacz — APPROVED** [review]

**Pull request merged**